crates/mem-pool/src/default_provider.rs

```diff
@@ -30,15 +30,42 @@ impl DefaultMemPoolProvider {
 
 #[async_trait]
 impl MemPoolProvider for DefaultMemPoolProvider {
+    // estimate next l2block timestamp
     async fn estimate_next_blocktime(&self) -> Result<Duration> {
-        // estimate next l2block timestamp
+        // Minus one second for first empty block
         const ONE_SECOND: Duration = Duration::from_secs(1);
+
         let rpc_client = &self.rpc_client;
-        let tip_block_hash = rpc_client.get_tip().await?.block_hash().unpack();
-        let opt_time = rpc_client.get_block_median_time(tip_block_hash).await?;
-        // Minus one second for first empty block
-        let minus_one_second = opt_time.map(|d| d - ONE_SECOND);
-        minus_one_second.ok_or_else(|| anyhow!("tip block median time not found"))
+        let tip_l1_block_hash_number = rpc_client.get_tip().await?;
+        let tip_l1_block_hash = tip_l1_block_hash_number.block_hash().unpack();
+        if let Some(median_time) = rpc_client.get_block_median_time(tip_l1_block_hash).await? {
+            return Ok(median_time - ONE_SECOND);
+        }
+
+        // tip l1 block hash is not on the current canonical chain, try parent block hash
+        // NOTE: Header doesn't include block hash
+        let mut l1_block_number = tip_l1_block_hash_number.number().unpack() + 1;
+        loop {
+            l1_block_number = l1_block_number.saturating_sub(1);
```

Comment:
IMO we should always retry `get_tip` RPC if we can't find tip block by tip hash. CKB node supposed to be updated to the new tip after the first call.

If we try to return median time by parent hash, we may use an outdated timestamp.